extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/responseheader/ResponseHeaderTest.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.resteasy.reactive.server.test.responseheader;
+
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.resteasy.reactive.server.common.Header;
+import io.quarkus.resteasy.reactive.server.common.ResponseHeader;
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.smallrye.mutiny.Uni;
+
+public class ResponseHeaderTest {
```

Comment:
We will also need tests to verify that `@ResponseHeader` is not taken into account when an exception is thrown.

Furthermore, we need to decide what the proper behavior is going to be when a user returns a `Response` or `RestResponse` which already give the ability to set the status